source/Calamari.Shared/Integration/Packages/NuGet/InternalNuGetPackageDownloader.cs

```diff
@@ -94,22 +97,67 @@ private void DownloadPackageAction(string packageId, IVersion version, Uri feedU
             // NuGet V3 feed 
             else if (IsHttp(feedUri.ToString()) && feedUri.ToString().EndsWith(".json", StringComparison.OrdinalIgnoreCase))
             {
-                NuGetV3Downloader.DownloadPackage(packageId, version, feedUri, feedCredentials, targetFilePath);
+                var timeout = GetHttpTimeout();
+                NuGetV3Downloader.DownloadPackage(packageId, version, feedUri, feedCredentials, targetFilePath, timeout);
             }
 
             // V2 feed
-            else 
+            else
             {
+                WarnIfHttpTimeoutHasBeenSet();
                 NuGetV2Downloader.DownloadPackage(packageId, version.ToString(), feedUri, feedCredentials, targetFilePath);
             }
 #else
             else
             {
+                WarnIfHttpTimeoutHasBeenSet();
                 NuGetV3LibDownloader.DownloadPackage(packageId, version, feedUri, feedCredentials, targetFilePath);
             }
 #endif
         }
 
+#if USE_NUGET_V2_LIBS
+        TimeSpan GetHttpTimeout()
+        {
+            const string expectedTimespanFormat = "c";
+            
+            // Equal to Timeout.InfiniteTimeSpan, which isn't available in net40
+            var defaultTimeout = new TimeSpan(0, 0, 0, 0, -1);
+            
+            var rawTimeout = variables.Get(KnownVariables.NugetHttpTimeout);
+            if (string.IsNullOrWhiteSpace(rawTimeout))
+            {
+                return defaultTimeout;
+            }
+
+            if (TimeSpan.TryParseExact(rawTimeout, expectedTimespanFormat, null, out var parsedTimeout))
+            {
+                return parsedTimeout;
+            }
+
+            var exampleTimespan = new TimeSpan(0, 0, 1, 0).ToString(expectedTimespanFormat);
+            
+            var message = $"The variable {KnownVariables.NugetHttpTimeout} couldn't be parsed as a timespan. " +
+                          $"Expected a value like '{exampleTimespan}' but received '{rawTimeout}'. " +
+                          $"Defaulting to '{defaultTimeout.ToString(expectedTimespanFormat)}'.";
+
+            Log.Warn(message);
+            return defaultTimeout;
+        }
+
+#endif
+        
+        void WarnIfHttpTimeoutHasBeenSet()
+        {
+            if (variables.IsSet(KnownVariables.NugetHttpTimeout))
+            {
+                Log.Warn(
+                    $"A Nuget HTTP timeout was set via the '{KnownVariables.NugetHttpTimeout}' variable. "
+                    + "This variable is not supported for this Nuget repository on this version of .NET."
```

Comment:
Consider inverting to say where it is supported rather than when it's not supported.
(not sure if I've got the right details here)
```suggestion
                    + "This variable is only supported when running on .NET Framework."
```